src/utils/Plotter.cpp

```diff
@@ -100,236 +63,336 @@ template <int D> void Plotter<D>::setSuffix(int t, const std::string &s) {
     closePlot();
     printout(20, std::endl);
 }
 
-/** Surface plot of a function
+/** Parametric plot of a function
 
-    Plots the function func in 2D between the endpoints A and B, to a file
-    named fname + file extension (".surf" as default). Endpoints and
-    the number of points must have been set at this point (default values are
-    1000 points between the boundaries of the function). The nPoints points
-    will be distributed evenly in the two dimensions.
+    Plots the function func parametrically along the vector A starting
+    from the origin O to a file named fname + file extension (".line"
+    as default).
 */
-template <int D> void Plotter<D>::surfPlot(const RepresentableFunction<D> &func, const std::string &fname) {
-    println(20, "--------Surface Plot----------");
-    if (not verifyRange()) MSG_ERROR("Zero range");
-    calcSurfCoordinates();
-    evaluateFunction(func);
+template <int D>
+void Plotter<D>::linePlot(const std::array<int, 1> &npts,
+                          const RepresentableFunction<D> &func,
+                          const std::string &fname) {
+    println(20, "----------Line Plot-----------");
     std::stringstream file;
-    file << fname << this->suffix[Plotter<D>::Surface];
-    openPlot(file.str());
-    writeSurfData();
-    closePlot();
+    file << fname << this->suffix[Plotter<D>::Line];
+    if (verifyRange(1)) {
+        Eigen::MatrixXd coords = calcLineCoordinates(npts[0]);
+        Eigen::VectorXd values = evaluateFunction(func, coords);
+        openPlot(file.str());
+        writeData(coords, values);
+        closePlot();
+    } else {
+        MSG_ERROR("Zero range");
+    }
     printout(20, std::endl);
 }
 
-/** Cubic plot of a function
+/** Parametric plot of a FunctionTree
 
-    Plots the function func in 3D between the endpoints A and B, to a file
-    named fname + file extension (".cube" as default). Endpoints and
-    the number of points must have been set at this point (default values are
-    1000 points between the boundaries of the function). The nPoints points
-    will be distributed evenly in the three dimensions.
+    Plots the function func parametrically along the vector A starting
+    from the origin O to a file named fname + file extension (".line"
+    as default).
 */
-template <int D> void Plotter<D>::cubePlot(const RepresentableFunction<D> &func, const std::string &fname) {
-    println(20, "----------Cube Plot-----------");
-    if (not verifyRange()) MSG_ERROR("Zero range");
-    calcCubeCoordinates();
-    evaluateFunction(func);
+template <int D>
+void Plotter<D>::linePlot(const std::array<int, 1> &npts, FunctionTree<D> &func, const std::string &fname) {
+    println(20, "----------Line Plot-----------");
     std::stringstream file;
-    file << fname << this->suffix[Plotter<D>::Cube];
-    openPlot(file.str());
-    writeCubeData();
-    closePlot();
+    file << fname << this->suffix[Plotter<D>::Line];
+    if (verifyRange(1)) {
```

Comment:
You could refactor to use the functions returning Eigen vectors inside these plotters.

Reply:
I removed the Eigen vector returners altogether. Unnecessary and never used.